lib/datadog/tracing/contrib/rack/middlewares.rb

```diff
@@ -304,7 +304,15 @@ def parse_url(env, original_env)
                        else
                          # normally REQUEST_URI starts at the path, but it
                          # might contain the full URL in some cases (e.g WEBrick)
-                         request_uri.sub(/^#{base_url}/, '')
+                         #
+                         # DEV: Use `request_uri.delete_prefix(base_url)`, supported in Ruby 2.5+
```

Comment:
I've received push back in the past with moving some refinements to a shared location.
A `String#delete_prefix` refinement already exists in `ddtrace`, but was kept close to its usage at the time: https://github.com/DataDog/dd-trace-rb/blob/53a90d91b2a531e282cb2be806c34c657d74d638/lib/datadog/tracing/distributed/trace_context.rb#L76-L87

This PR could be simplified to use the refinement, but that refinement declaration would have to be moved to a more shareable location than it is today.

This PR becomes
```diff
- request_uri.sub(/^#{base_url}/, '')
+ request_uri.delete_prefix(base_url)
```
with the refinement.

Reply:
I extracted it to its own method.